pkg/table/tables/tables.go

```diff
@@ -670,7 +670,7 @@ func (t *TableCommon) UpdateRecord(ctx context.Context, sctx table.MutateContext
 			return err
 		}
 	}
-	colSize := make(map[int64]int64, len(t.Cols()))
+	colSize := make([]variable.ColSize, len(t.Cols()))
```

Comment:
Nice catch, we could check if there similar implemetations in other performance critical paths.